examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+    storage::Flush,
+};
+use ink_lang::contract;
+use ink_model::EnvHandler;
+use scale::{Encode, Decode};
+
+pub type Result<T, E> = core::result::Result<T, E>;
+pub type TokenId = u32;
+
+#[derive(Encode, Decode)]
+#[cfg_attr(feature = "ink-generate-abi", derive(Metadata))]
+struct Counter {
+    value: storage::Value<u32>,
+}
+
+impl Counter {
+    pub fn increase(&mut self) {
+        self.value += 1;
+    }
+
+    pub fn decrease(&mut self) {
+        self.value -= 1;
+    }
+
+    pub fn get(&self) -> u32 {
+        *self.value
+    }
+}
+
+impl Flush for Counter {
+	fn flush(&mut self) {}
+}
+
+contract! {
+    #![env = DefaultSrmlTypes]
+
+    // Event deposited when a token transfer occurs
+    event Transfer {
+        from: AccountId,
+        to: AccountId,
+        id: TokenId,
+    }
+
+    // Event deposited when an approval occurs
+    event Approval {
+        owner: AccountId,
+        to: AccountId,
+        id: TokenId,
+    }
+
+    /// The storage items for a typical ERC721 token implementation.
+    struct Erc721 {
+        /// The total supply.
+        token_owner: storage::HashMap<TokenId, AccountId>,
+        token_approvals: storage::HashMap<TokenId, AccountId>,
+        owned_tokens_count: storage::HashMap<AccountId, Counter>,
+        all_tokens: storage::Vec<TokenId>,
+    }
+
+    impl Deploy for Erc721 {
+        fn deploy(&mut self, name: u8, symbol: u8) {
+        }
+    }
+
+    impl Erc721 {
+        pub(external) fn get_total_supply(&self) -> u32 {
+            let total_supply = self.all_tokens.len();
+            env.println(&format!("Erc20::total_supply = {:?}", total_supply));
+            total_supply
+        }
+        
+        pub(external) fn get_balance(&self, owner: AccountId) -> u32 {
+            let balance = self.balance_of(&owner);
+            env.println(&format!("Erc721::balance_of(owner = {:?}) = {:?}", owner, balance));
+            balance
+        }
+
+        pub(external) fn get_owner(&self, id: u32) -> AccountId {
+            let owner = self.owner_of(&id);
+            env.println(&format!("Erc721::owner_of(token = {:?}) = {:?}", id, owner));
+            owner
+        }
+    }
+
+    impl Erc721 {
+        fn approve(&mut self, env: &mut EnvHandler<ink_core::env::ContractEnv<DefaultSrmlTypes>>, to: &AccountId, id: &TokenId) -> Result<(), &'static str> {
```

Comment:
Please create a type alias `type EnvHandler = EnvHandler<ink_core::env::ContractEnv<DefaultSrmlTypes>>` or similar to end this abomination.
I am so glad we are getting rid of this in the upcoming `ink_lang2` codegen.